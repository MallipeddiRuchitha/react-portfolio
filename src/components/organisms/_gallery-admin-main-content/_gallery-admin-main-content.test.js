import React from 'react';
import { shallow, mount } from 'enzyme';
import GalleryAdmin from './_gallery-admin-main-content';
import AddImageForm from "../../molecules/_add-image-form/_add-image-form";
import EditImageForm from "../../molecules/edit-image-form/_edit-image-form";
import EditIcon from "@material-ui/icons/Edit";
import DeleteIcon from "@material-ui/icons/Delete";
import ImageCard from "../_image-card/_image-card";
import SvgIcon from '@material-ui/icons/Edit';
//import EditImageForm from '../../molecules/edit-image-form/_edit-image-form';
describe("GalleryAdmin component", () => {
    test("Matches the snapshot", () => {
        const galleryAdmin = shallow(<GalleryAdmin />);
        expect(galleryAdmin).toMatchSnapshot();

    })
    test("GalleryAdmmin is rendered", () => {
        const galleryAdmin = shallow(<GalleryAdmin />);

        expect(galleryAdmin.find('#galleryAdmin').exists()).toEqual(true);
    });
    test("initially Add Image Form is not rendered", () => {
        const galleryAdmin = mount(<GalleryAdmin />);

        expect(galleryAdmin.find(AddImageForm).exists()).toEqual(false);
        galleryAdmin.unmount();
    });
    test("Adding Image ,on clicking Add button Add Image Form should be rendered", () => {
        const galleryAdmin = mount(<GalleryAdmin />);
        expect(galleryAdmin.find(ImageCard).length).toEqual(6);
        const button = galleryAdmin.find("#button").last();
        button.simulate('click');
        //expect(button.text()).toEqual(' Add');
        expect(galleryAdmin.find(AddImageForm).exists()).toEqual(true);
        galleryAdmin.find('#Name').last().simulate('change', { target: { value: 'forest71' } });
        galleryAdmin.find('#Url').last().simulate('change', { target: { value: 'https://www.birmingham.ac.uk/Images/News/Forest-900.jpg' } });
        galleryAdmin.find('#Name').last().simulate('change', { target: { value: 'forest7' } });
        //console.log("11111111111",galleryAdmin.debug())
        expect(galleryAdmin.find("#button").last().props()['disabled']).toBe(false)
        galleryAdmin.find("#button").last().simulate('click');

        expect(galleryAdmin.find(ImageCard).length).toEqual(7);
        galleryAdmin.unmount();
    });

    test("Editing ImageCard ", () => {
        const galleryAdmin = mount(<GalleryAdmin />);
        const imageCard = galleryAdmin.find(ImageCard).last();
        expect(galleryAdmin.find(ImageCard).last().text()).toEqual("Forest6");
        imageCard.find(SvgIcon).first().simulate('click');
        //console.log("11111111111",galleryAdmin.find(ImageCard).last().debug())   
        galleryAdmin.find(ImageCard).last().find('#Name').last().simulate('change', { target: { value: 'forest612' } });

        galleryAdmin.find(ImageCard).last().find("#button").last().simulate('click');

        expect(galleryAdmin.find(ImageCard).last().text()).toEqual("forest612");
        galleryAdmin.unmount();
    });
    test("Deleting Image Card", () => {
        const galleryAdmin = mount(<GalleryAdmin />);
        expect(galleryAdmin.find(ImageCard).length).toEqual(6);
        const imageCard = galleryAdmin.find(ImageCard).last();        
        imageCard.find("#deleteIcon").last().simulate('click');
        expect(galleryAdmin.find(ImageCard).length).toEqual(5);
        galleryAdmin.unmount();
    });








});