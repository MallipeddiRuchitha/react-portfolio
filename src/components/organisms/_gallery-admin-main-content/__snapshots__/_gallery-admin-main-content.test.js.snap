// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GalleryAdmin component Matches the snapshot 1`] = `
<div
  className="makeStyles-styleGalleryAdminMainContent-1"
  id="galleryAdmin"
>
  <div
    className="makeStyles-button-4"
  >
    <CustomButton
      onClick={[Function]}
      value="Add"
    />
  </div>
  <div
    className="makeStyles-styleGrid-2"
  >
    <ImageCard
      handleDelete={[Function]}
      handleEdit={[Function]}
      image1={
        Object {
          "id": "1",
          "name": "Forest1",
          "url": "https://www.birmingham.ac.uk/Images/News/Forest-900.jpg",
        }
      }
      key="1"
    />
    <ImageCard
      handleDelete={[Function]}
      handleEdit={[Function]}
      image1={
        Object {
          "id": "2",
          "name": "Forest2",
          "url": "https://encrypted-tbn0.gstatic.com/images?q=tbn%3AANd9GcQIWGOJcO5wb_qh4welWgFUgsZPHXMetI-iVg&usqp=CAU",
        }
      }
      key="2"
    />
    <ImageCard
      handleDelete={[Function]}
      handleEdit={[Function]}
      image1={
        Object {
          "id": "3",
          "name": "Forest3",
          "url": "https://miro.medium.com/max/8576/0*zBu6EBAwjXXXHz-z",
        }
      }
      key="3"
    />
    <ImageCard
      handleDelete={[Function]}
      handleEdit={[Function]}
      image1={
        Object {
          "id": "4",
          "name": "Forest4",
          "url": "https://api.timeforkids.com/wp-content/uploads/2019/09/final-cover-forest.jpg",
        }
      }
      key="4"
    />
    <ImageCard
      handleDelete={[Function]}
      handleEdit={[Function]}
      image1={
        Object {
          "id": "5",
          "name": "Forest5",
          "url": "https://www.orfonline.org/wp-content/uploads/2020/03/India-forests-1280x720.jpg",
        }
      }
      key="5"
    />
    <ImageCard
      handleDelete={[Function]}
      handleEdit={[Function]}
      image1={
        Object {
          "id": "6",
          "name": "Forest6",
          "url": "https://encrypted-tbn0.gstatic.com/images?q=tbn%3AANd9GcQfbbQr8RYShw7zga6H2q9kTF78VpUMq22VGA&usqp=CAU",
        }
      }
      key="6"
    />
  </div>
</div>
`;
